sympy/core/tests/test_function.py

```diff
@@ -1100,12 +1100,14 @@ def test_function_assumptions():
     x = Symbol('x')
     f = Function('f')
     f_real = Function('f', real=True)
+    f_real_inherit = Function(Symbol('f', real=True))
 
     assert f != f_real
     assert f(x) != f_real(x)
 
     assert f(x).is_real is None
     assert f_real(x).is_real is True
+    assert f_real_inherit(x).is_real is True and str(f_real_inherit) == 'f'
```

Comment:
```
>>> f=Function('f')
>>> f.name
'f'
```
`f_real_inherit.name == 'f'` should work, else something is broken?

Reply:
Your very right, I thought for a second that it wasn't supposed to be a string because I didn't think I'd made a change that would affect the name. But sure enough, there's a line in the constructor that sets the function name equal to what now can be a symbol. Fixed in the latest commit, hopefully, that also takes care of the failing CI tests.